### PR TITLE
Feature persistency service

### PR DESCRIPTION
**Pull request by gmojunior**

### CODE REVIEW

File: PontoFacil/PontoFacil/Services/PersistencyService.cs [hunk 1]
```diff
@@ -1,12 +1,101 @@
...
+        public static PersistencyService getInstance()
+        {
+            if (persistencyService == null)
+                persistencyService = new PersistencyService();
+            return persistencyService;
+        }
+        #endregion
+
+        #region Methods
+        public async Task persist()
```
Comment (tarluna-zz) [inline PontoFacil/PontoFacil/Services/PersistencyService.cs]: Methods should start with uppercase letter

File: PontoFacil/PontoFacil/Services/PersistencyService.cs [hunk 2]
```diff
@@ -1,12 +1,101 @@
...
+        {
+            await Task.Run(() =>
+                File.WriteAllText(DATABASE_PATH, JsonConvert.SerializeObject(persistencyService))
+            );
+        }
+
+        public void restore()
```
Comment (tarluna-zz) [inline PontoFacil/PontoFacil/Services/PersistencyService.cs]: Methods should start with uppercase letter

File: PontoFacil/PontoFacil/Services/PersistencyService.cs [hunk 3]
```diff
@@ -1,12 +1,101 @@
...
+            {
+                Console.WriteLine(fileNotFoundException.Message);
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e.Message);
+            }
+        }
+
+        public void saveClockIn(ClockIn clockIn)
```
Comment (tarluna-zz) [inline PontoFacil/PontoFacil/Services/PersistencyService.cs]: Methods should start with uppercase letter

File: PontoFacil/PontoFacil/Services/PersistencyService.cs [hunk 4]
```diff
@@ -1,12 +1,101 @@
...
+        {
+            persistencyService.clockInList.Add(clockIn);
+        }
+
+        public void savePlanning(Planning planning)
```
Comment (tarluna-zz) [inline PontoFacil/PontoFacil/Services/PersistencyService.cs]: Methods should start with uppercase letter

File: PontoFacil/PontoFacil/Services/PersistencyService.cs [hunk 5]
```diff
@@ -1,12 +1,101 @@
...
+        {
+            persistencyService.MyPlanning = planning;
+        }
+
+        public void saveProfile(Profile profile)
```
Comment (tarluna-zz) [inline PontoFacil/PontoFacil/Services/PersistencyService.cs]: Methods should start with uppercase letter

**Review by tarluna-zz — COMMENTED** [review]
Please, check my comments

File: PontoFacil/PontoFacil/Services/PersistencyService.cs [hunk 6]
```diff
@@ -1,12 +1,101 @@
...
+using Windows.Storage;
 
 namespace PontoFacil.Services
 {
-    class PersistencyService
+    class PersistencyService : IPersistencyService
     {
+        #region Properties
+        private readonly string DATA_FILE_NAME = "PontoFacilData.txt";
+        private readonly string PATH_SEPARTOR = @"\";
```
Comment (lhohenrique) [inline PontoFacil/PontoFacil/Services/PersistencyService.cs]: Just a minor change. The property name should be PATH_SEPARATOR

File: PontoFacil/PontoFacil/Services/PersistencyService.cs [hunk 7]
```diff
@@ -1,12 +1,101 @@
...
+
+        #region Construcutor
+        private PersistencyService()
+        {
+            DATABASE_PATH = DATABASE_FOLDER + PATH_SEPARTOR + DATA_FILE_NAME;
+
+            this.clockInList = new List<ClockIn>();
+        }
+
+        public static PersistencyService getInstance()
```
Comment (tarluna-zz) [inline PontoFacil/PontoFacil/Services/PersistencyService.cs]: Methods should start with uppercase letter

**Review by tarluna-zz — CHANGES_REQUESTED** [review]

**Pull request merged**